fhir-path/src/test/java/com/ibm/fhir/path/test/BetweenFunctionTest.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Test
+    public void testBetweenFunction4() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@1970-01-01, @2020-01-01, 'days')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(18262), "days"));
+    }
+
+    @Test
+    public void testBetweenFunction5() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@T00:00:00.000, @T12:34:56.789, 'hours')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(12), "hours"));
+    }
+
+    @Test
+    public void testBetweenFunction6() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@T00:00:00.000, @T12:34:56.789, 'minutes')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(754), "minutes"));
+    }
+
+    @Test
+    public void testBetweenFunction7() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@T00:00:00.000, @T12:34:56.789, 'seconds')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(45296), "seconds"));
+    }
+
+    @Test
+    public void testBetweenFunction8() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@T00:00:00.000, @T12:34:56.789, 'milliseconds')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(45296789), "milliseconds"));
+    }
+
+    @Test
+    public void testBetweenFunction9() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@2020-01-01, @1970-01-01, 'years')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(-50), "years"));
+    }
+
+    @Test
+    public void testBetweenFunction10() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        Collection<FHIRPathNode> result = evaluator.evaluate("between(@1970-01-01, @2020-01-01T00:00:00Z, 'years')");
+        assertEquals(getQuantityValue(result), quantityValue(new BigDecimal(50), "years"));
+    }
+
+    @Test
+    public void testBetweenFunction11() throws Exception {
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        try {
+            evaluator.evaluate("between(@1970-01-01T00:00:00Z, @2020-01-01, 'years')");
+            fail();
+        } catch (Exception e) {
+            assertTrue(e instanceof FHIRPathException);
+            assertTrue(e.getCause() instanceof DateTimeException);
+        }
+    }
```

Comment:
could use TestNG expectedExceptions, but this is fine

Reply:
I prefer this approach because I can check exception type and message